src/Microsoft.AspNetCore.Server.IntegrationTesting/Common/TestUriHelper.cs

```diff
@@ -36,6 +36,12 @@ public static int FindFreePort(int initialPort)
         {
             using (var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
             {
+                // Set linger to true, but the linger time to zero, which forces the socket to shut down immediately when closed
+                // The default behavior is that linger is off, which actually means that the socket may sit for a small amount
+                // of time before being fully closed and reset.
+                // More here: https://msdn.microsoft.com/en-us/library/windows/desktop/ms737582(v=vs.85).aspx
+                socket.LingerState = new LingerOption(enable: true, seconds: 0);
```

Comment:
I wouldn't add this unless we know it will help.  I'm not sure it even matters if no client ever connected to the socket.  I'm testing this now.